### ISSUE TITLE
Question: animating something that doesn't get removed/destroyed

### ISSUE THREAD

**Issue by dmitriid** [snabbdom/snabbdom#215]
tl;dr: I have two similar forms. I want the forms to disappear and appear when switching between them. Snabbdom only patches the insides of the form, so `remove`, `delayed` etc. are either not triggered, or triggered only once. How do I do it?

Long version:

I have two very similar forms on the page. Something like this:

```js
export default LoginForm() {
  return h('form', [
    h('input', {attrs: {name: 'username'}}),
    h('input', {attrs: {name: 'password'}})
  ])
}
```
```js
export default RegisterForm() {
  return h('form', [
    h('input', {attrs: {name: 'username'}}),
    h('input', {attrs: {name: 'password'}}),
    h('input', {attrs: {name: 'activationCode'}})
  ])
}
```

And I display it on the page not dissimilar to this:

```
function mount(element, view) {
  let tree = h('span')
  patch(element, tree)

  function render() {
    const newTree = view()
    patch(tree, newTree)
    tree = newTree
  }

  autorun(() => render())
}

mount(
  document.getElementById('app'),
  () => AppState.currentState === 'users.register' ? RegisterForm() : LoginForm()
)
```

So, when the `currentState` changes, a new view is produced and patched into the DOM.

Obviously snabbdom is smart and when I switch between forms it only adds/removes the extra field (there are also some texts that change, but that's it).

Therefore, if add the following style to my forms, the style is only triggered once, when the page is loaded:

```
      style: {
        opacity:    '0',
        transition: 'opacity 1s',
        delayed:    { opacity: '1' },  // only invoked once, when page is loaded
        remove:     { opacity: '0' } // never invoked because the form isn't removed
      }
```

So, my question is: what would be the best way to actually "force" switching between the forms?

**Comment by TylorS** [comment]
@dmitriid You could add keys to your forms. `form({ key: 1 })` for example.

**Comment by dmitriid** [comment]
Perfect! There's some weird interaction between how forms appear and disappear now, but this I can probably solve :)

Thank you!
